modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/tooling/ExtensionConnectivityTestingStrategy.java

```diff
@@ -52,21 +57,38 @@ void setMuleContext(MuleContext muleContext) {
   @Override
   public ConnectionValidationResult testConnectivity(Object connectivityTestingObject) {
     try {
-      ConnectionProvider connectionProvider =
-          ((ConnectionProviderResolver) connectivityTestingObject).resolve(getInitialiserEvent(muleContext));
-      Object connection = connectionProvider.connect();
-      return connectionProvider.validate(connection);
+      if (connectivityTestingObject instanceof ConnectionProviderResolver) {
+        ConnectionProvider connectionProvider =
+            ((ConnectionProviderResolver) connectivityTestingObject).resolve(getInitialiserEvent(muleContext));
+        return validateConnectionOverConnectionProvider(connectionProvider);
+      } else {
+        ConfigurationProvider configurationProvider = (ConfigurationProvider) connectivityTestingObject;
+        ConfigurationInstance configurationInstance = configurationProvider.get(getInitialiserEvent(muleContext));
+        configurationInstance.getConnectionProvider();
+        if (configurationInstance.getConnectionProvider().isPresent()) {
+          return validateConnectionOverConnectionProvider(configurationInstance.getConnectionProvider().get());
+        } else {
+          throw new MuleRuntimeException(createStaticMessage("The component does not support connectivity testing"));
```

Comment:
Because ConnectionValidationResult is only valid when testing the connection was possible. In that case there's a bug in the client code since its requesting connection testing over an object that does not support it.